app/src/main/java/com/duckduckgo/app/globalprivacycontrol/GlobalPrivacyControlManager.kt

```diff
@@ -17,18 +17,43 @@
 package com.duckduckgo.app.globalprivacycontrol
 
 import android.content.Context
+import android.net.Uri
 import android.webkit.WebView
 import androidx.annotation.UiThread
 import com.duckduckgo.app.browser.R
+import com.duckduckgo.app.global.UriString
+import com.duckduckgo.app.global.domain
 import com.duckduckgo.app.settings.db.SettingsDataStore
-import javax.inject.Inject
 
-interface GlobalPrivacyControlInjector {
+interface GlobalPrivacyControl {
     fun injectDoNotSellToDom(webView: WebView)
+    fun isGpcActive(): Boolean
+    fun getHeaders(): Map<String, String>
+    fun shouldAddHeaders(url: Uri): Boolean
 }
 
-class GlobalPrivacyControlInjectorJs @Inject constructor(private val appSettingsPreferencesStore: SettingsDataStore) : GlobalPrivacyControlInjector {
+class GlobalPrivacyControlManager(private val appSettingsPreferencesStore: SettingsDataStore) : GlobalPrivacyControl {
     private val javaScriptInjector = JavaScriptInjector()
+    private val headerConsumers = listOf(
+        "nytimes.com",
+        "globalprivacycontrol.org",
+        "global-privacy-control.glitch.me"
+    )
+
+    override fun shouldAddHeaders(url: Uri): Boolean {
```

Comment:
Consider adding tests for this method. Also test for subdomains, i.e. `static.nytimes.com`.

Reply:
Also on this note. This check only happens in the `WebViewRequestInterceptor` but we are also adding the headers in the `BrowserTabViewModel`. Should this check be added in the ViewModel `getUrlHeaders()` method? (line 531)

Reply:
Just gonna leave this as it is for now :D